### PR TITLE
[Bugfix][Spec Decode] Disable AutoBlockify for rejection sampling

### PR DESCRIPTION
**Pull request by jiajinzhu2**
### What this PR does / why we need it?

Fixes an Ascend Triton correctness issue in rejection sampling. AutoBlockify can corrupt the max-with-index reductions used by `_compute_block_stats_kernel` and `_resample_kernel`, which can produce an incorrect argmax/token ID even when the input logits are finite.

This patch opts both rejection-sampling kernels out of AutoBlockify with `has_auto_blockify_blacklist_op=True`. It also re-enables the existing DeepSeek V4 MTP eager test by removing its temporary skip marker.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- `python -m py_compile vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py`
- DeepSeek V4 MTP eager diagnostic soak: 448 consecutive iterations completed on four Ascend NPUs without an invalid token ID or NPU device error before the run was manually stopped to reload the clean patch.

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_eager
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33062712545) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_eager
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33066987387) for details.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_eager
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33072474520) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_eager
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33133673399) for details.
[Bot]: e2e command failed.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33135680674) for details.
[Bot]: e2e command failed.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33136891039) for details.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33143290895) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33146806365) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33149392487) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33151322893) for details.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33151841688) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_full_decode_only
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33226644873) for details.
[Bot]: e2e command completed successfully.

**Comment by jiajinzhu2** [comment]
/e2e tests/e2e/pull_request/four_card/model_runner_v2/test_deepseek_v4.py::test_deepseek_v4_mtp_eager
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/33460305448) for details.
[Bot]: e2e command completed successfully.

File: vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py [hunk 1]
```diff
@@ -411,6 +411,10 @@ def rejection_sample(
         num_speculative_steps,
         BLOCK_SIZE=VOCAB_BLOCK_SIZE,
         HAS_DRAFT_LOGITS=has_draft_logits,
+        # The kernel contains a max-with-index reduction. Keep AutoBlockify
```
Comment (drslark) [inline vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py]: Add a TODO.

**Review by drslark — CHANGES_REQUESTED** [review]

**Review by drslark — APPROVED** [review]
LGTM.

**Review by AuroraEmiya — APPROVED** [review]

**Review by lilinsiman — APPROVED** [review]

**Review by kunpengW-code — APPROVED** [review]
